include/swift/SIL/SILBuilder.h

```diff
@@ -480,9 +482,10 @@ class SILBuilder {
                                bool reflection = false,
                                bool usesMoveableValueDebugInfo = false
 #ifndef NDEBUG
-                               , bool skipVarDeclAssert = false
+                               ,
+                               bool skipVarDeclAssert = false,
 #endif
-                               ) {
+                               bool hasPointerEscape = false) {
     llvm::SmallString<4> Name;
```

Comment:
This is incorrect when `NDEBUG` is defined.

It will expand to the following with NDEBUG defined:

```
                               bool reflection = false,
                               bool usesMoveableValueDebugInfo = false
                               bool hasPointerEscape = false) {
```

Which is missing the `,` at the end of the `usesMoveableValueDebugInfo`.

I think the right format should be:

```
                               bool usesMoveableValueDebugInfo = false,
#ifndef NDEBUG
                               bool skipVarDeclAssert = false,
#endif
                               bool hasPointerEscepe = false) {
```

Reply:
I just realized that the calls in many places assume that the parameter is always there and do not seem to change when `NDEBUG` is defined or not. The code of the function also do not check for `NDEBUG` when using `skipVarDeclAssert`. It might be better to remove the preprocessor conditional completely instead of fixing the format.

Reply:
Thanks @drodriguez! I don't think it makes sense for an individual parameter to be under NDEBUG. If it's unused in the implementation, that's easily handled with `(void)skipVarDeclAssert`

Reply:
Here is the PR - https://github.com/apple/swift/pull/66304